Index: Gemfile
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Gemfile b/Gemfile
--- a/Gemfile	(revision 628e73f1a386667295d122d6e351dcbbab3d3699)
+++ b/Gemfile	(date 1626877064832)
@@ -17,6 +17,7 @@
 gem 'turbolinks', '~> 5'
 # Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder
 gem 'jbuilder', '~> 2.7'
+gem 'devise', '~> 4.2'
 # Use Redis adapter to run Action Cable in production
 # gem 'redis', '~> 4.0'
 # Use Active Model has_secure_password
Index: config/routes.rb
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/config/routes.rb b/config/routes.rb
--- a/config/routes.rb	(revision 628e73f1a386667295d122d6e351dcbbab3d3699)
+++ b/config/routes.rb	(date 1626881216102)
@@ -1,3 +1,5 @@
 Rails.application.routes.draw do
+  devise_for :users
   # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  root 'home#index'
 end
Index: app/views/home/index.html.erb
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/views/home/index.html.erb b/app/views/home/index.html.erb
new file mode 100644
--- /dev/null	(date 1626943539399)
+++ b/app/views/home/index.html.erb	(date 1626943539399)
@@ -0,0 +1,7 @@
+<h1>Main page</h1>
+<% if user_signed_in? %>
+  <h3>Welcome!</h3>
+  <%= link_to 'Sign Out', destroy_user_session_path, method: :delete %>
+<% else %>
+  <%= link_to 'Sign Up', new_user_registration_path %> <span> or if you have account </span> <%= link_to 'Sign In', new_user_session_path %>
+<% end %>
\ No newline at end of file
Index: Gemfile.lock
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Gemfile.lock b/Gemfile.lock
--- a/Gemfile.lock	(revision 628e73f1a386667295d122d6e351dcbbab3d3699)
+++ b/Gemfile.lock	(date 1626877076075)
@@ -62,6 +62,7 @@
       zeitwerk (~> 2.3)
     addressable (2.8.0)
       public_suffix (>= 2.0.2, < 5.0)
+    bcrypt (3.1.16)
     bindex (0.8.1)
     bootsnap (1.7.5)
       msgpack (~> 1.0)
@@ -78,6 +79,12 @@
     childprocess (3.0.0)
     concurrent-ruby (1.1.9)
     crass (1.0.6)
+    devise (4.8.0)
+      bcrypt (~> 3.0)
+      orm_adapter (~> 0.1)
+      railties (>= 4.1.0)
+      responders
+      warden (~> 1.2.3)
     erubi (1.10.0)
     ffi (1.15.3)
     globalid (0.4.2)
@@ -104,6 +111,7 @@
     nokogiri (1.11.7)
       mini_portile2 (~> 2.5.0)
       racc (~> 1.4)
+    orm_adapter (0.5.0)
     public_suffix (4.0.6)
     puma (5.3.2)
       nio4r (~> 2.0)
@@ -146,6 +154,9 @@
     rb-inotify (0.10.1)
       ffi (~> 1.0)
     regexp_parser (2.1.1)
+    responders (3.0.1)
+      actionpack (>= 5.0)
+      railties (>= 5.0)
     rubyzip (2.3.2)
     sass-rails (6.0.0)
       sassc-rails (~> 2.1, >= 2.1.1)
@@ -177,6 +188,8 @@
     turbolinks-source (5.2.0)
     tzinfo (2.0.4)
       concurrent-ruby (~> 1.0)
+    warden (1.2.9)
+      rack (>= 2.0.9)
     web-console (4.1.0)
       actionview (>= 6.0.0)
       activemodel (>= 6.0.0)
@@ -205,6 +218,7 @@
   bootsnap (>= 1.4.4)
   byebug
   capybara (>= 3.26)
+  devise (~> 4.2)
   jbuilder (~> 2.7)
   listen (~> 3.3)
   puma (~> 5.0)
Index: app/views/layouts/application.html.erb
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/views/layouts/application.html.erb b/app/views/layouts/application.html.erb
--- a/app/views/layouts/application.html.erb	(revision 628e73f1a386667295d122d6e351dcbbab3d3699)
+++ b/app/views/layouts/application.html.erb	(date 1626881020227)
@@ -11,6 +11,7 @@
   </head>
 
   <body>
+    <%= render 'layouts/alerts' %>
     <%= yield %>
   </body>
 </html>
Index: app/controllers/home_controller.rb
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/controllers/home_controller.rb b/app/controllers/home_controller.rb
new file mode 100644
--- /dev/null	(date 1626878582308)
+++ b/app/controllers/home_controller.rb	(date 1626878582308)
@@ -0,0 +1,4 @@
+class HomeController < ApplicationController
+  def index
+  end
+end
Index: app/views/layouts/_alerts.html.erb
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/app/views/layouts/_alerts.html.erb b/app/views/layouts/_alerts.html.erb
new file mode 100644
--- /dev/null	(date 1626881020230)
+++ b/app/views/layouts/_alerts.html.erb	(date 1626881020230)
@@ -0,0 +1,2 @@
+<p class="notice"><%= notice %></p>
+<p class="alert"><%= alert %></p>
Index: config/environments/development.rb
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/config/environments/development.rb b/config/environments/development.rb
--- a/config/environments/development.rb	(revision 628e73f1a386667295d122d6e351dcbbab3d3699)
+++ b/config/environments/development.rb	(date 1626878522537)
@@ -73,4 +73,5 @@
 
   # Uncomment if you wish to allow Action Cable access from any origin.
   # config.action_cable.disable_request_forgery_protection = true
+  config.action_mailer.default_url_options = { host: 'localhost', port: 3000 }
 end
